doc/Language/iterating.pod6

```diff
@@ -4,136 +4,321 @@
+attempts to fetch more data, and behavior for doing so is undefined. The only
+valid use of the sentinel value C<IterationEnd> in a program is identity
+comparison (using L<C<=:=>|/routine/=:=>) with the result of a method in the
+iterator API. Any other behavior is undefined and implementation dependent.
+
+Thus, the manual iteration can be rewritten as a loop:
+
+=begin code
+my @languages = <perl python haskell ruby>;
+
+my $iter = @languages.iterator;
+loop {
+    my \value = $iter.pull-one;
+    last if value =:= IterationEnd;
+    say value;
+}
+=end code
+
+Since C<IterationEnd> is a constant, it's important that the variable we compare
+against it be bound, and not assigned. Hence, the use of the sigilless variable
+C<value> in the previous example.
 
 =head1 How to iterate: contextualizing and topic variables
 
-C<for> and other loops place the item produced in every iteration into the
+Loops, such as C<for>, place the item produced in every iteration into the
 L<topic variable C<$_>|/language/variables#index-entry-topic_variable>, or
 capture them into the variables that are declared along with the block. These
 variables can be directly used inside the loop, without needing to declare them,
 by using the
 L<C<^> twigil|/syntax/$CIRCUMFLEX_ACCENT#(Traps_to_avoid)_twigil_^>.
 
-Implicit iteration occurs when using the L<sequence operator|/language/operators#index-entry-..._operators>.
+Implicit iteration occurs when using the L<sequence operator|/language/operators#index-entry-..._operators>:
 
-    say 1,1,1, { $^a²+2*$^b+$^c } … * > 300; # OUTPUT: «(1 1 1 4 7 16 46 127 475)
+    say 1,1,1, { $^a² + 2*$^b + $^c } … * > 300; # OUTPUT: «(1 1 1 4 7 16 46 127 475)
 
 The generating block is being run once while the condition to finish the
 sequence, in this case the term being bigger than 300, is not met. This has the
-side effect of running a loop, but also creating a list that is output.
+side effect of both running a loop and creating a list.
 
-This can be done more systematically through the use of the L<C<gather/take>
-blocks|/syntax/gather take>, which are a different kind of iterating construct
-that instead of running in sink context, returns an item every iteration. This
+This can be done more systematically through the use of
+L<C<gather/take> blocks|/syntax/gather take>, which are a different kind of
+iterating construct that instead of running in sink context, returns an item
+every iteration. This
 L<Advent Calendar tutorial|https://perl6advent.wordpress.com/2009/12/23/day-23-lazy-fruits-from-the-gather-of-eden/>
-explains use cases for this kind of loops; in fact, gather is not so much a
+explains use cases for this kind of loops; in fact, C<gather> is not so much a
 looping construct, but a statement prefix that collects the items produced by
 C<take> and creates a list out of them.
 
-=head1 C<Classic> loops and why we do not like them
+=head1 Classical loops and why we do not like them
```

Comment:
I think "Classic" is better here. "Classic-style" if you preffer, or in italics instead of monospace.